sdk/core/Azure.Core/samples/ProtocolMethods.md

```diff
@@ -0,0 +1,139 @@
+# C# Azure SDK Clients that Contain Protocol Methods
+
+## Introduction
+
+Most Azure SDK Clients expose methods that take 'model' parameters, C# classes which to the input\output of a REST JSON API call.
+
+Imagine a service that stores information about pets, with a GetDog and SetDog pair of operations. 
+
+Pets are represented in the request by a JSON hash:
+
+```json
+{
+        "name": "Buddy",
+        "id": 2,
+        "color": "Brown"
+}
+```
+
+An API using models might be:
+
+```csharp
+        public class Pet
+        {
+                string Name { get; }
+                int Id { get; }
+                string Color { get; }
+        }
+
+        Task<Response<Pet>> GetDog(string dogName, CancellationToken cancellation = default);
+        Task<Response> SetDog(Pet dog, CancellationToken cancellation = default);
```

Comment:
Nit: Add the 'Async' suffix on these task-returning, cancellation-token-taking methods.